core/arch/arm/kernel/rpc_io_i2c.c

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 Foundries Ltd <jorge@foundries.io>
+ *
+ */
+#include <kernel/rpc_io_i2c.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <string.h>
+
+#define I2C_BUFFER_LENGTH 512 /* enough for most use cases */
+static struct mobj *mobj;
+
+TEE_Result rpc_io_i2c_transfer(struct rpc_i2c_request *req, size_t *bytes)
+{
+	TEE_Result res = TEE_SUCCESS;
+	struct thread_param p[3] = { };
+
+	if (!req)
```

Comment:
is this really needed?

Reply:
not sure what to say to this. it is not needed if nobody calls the function with a null pointer...seems on the same line of thought of having to initialize all the stack variables even when there is only one.

Reply:
It is ok as is but I think `assert(req);` would be enough.

Reply:
I think you are right. ok